packages/gosu.rb

```diff
@@ -0,0 +1,27 @@
+require 'package'
+
+class Gosu < Package
+  description 'Simple Go-based setuid+setgid+setgroups+exec'
+  homepage 'https://github.com/tianon/gosu'
+  version '1.14'
+  license 'Apache-2.0'
+  compatibility 'all'
+  source_url 'SKIP'
+
+  def self.install
+    case ARCH
+    when 'aarch64', 'armv7l'
+      url = 'https://github.com/tianon/gosu/releases/download/1.14/gosu-armhf'
```

Comment:
Use `source_url` and install the file with `Dir['gosu-*']`?

Reply:
No sure what you mean.  Is that a simplified way to install?

Reply:
That won't work because the binaries are not in a compressed format.  They simply need to be downloaded and intsalled.

Reply:
> Not sure what you mean. Is that a simplified way to install?

Kind of :) It takes lesser lines of code to achieve
```ruby
source_url ({
  x86_64: '...',
  ...
})

def self.install
  FileUtils.install Dir['gosu-*'], ...
end
```

> That won't work because the binaries are not in a compressed format. They simply need to be downloaded and intsalled.
The non-archive file support for `source_url` has been added in #6743

Reply:
It doesn't work. :)  See below:
```
require 'package'

class Gosu < Package
  description 'Simple Go-based setuid+setgid+setgroups+exec'
  homepage 'https://github.com/tianon/gosu'
  version '1.14'
  license 'Apache-2.0'
  compatibility 'all'
  source_url ({
    aarch64: 'https://github.com/tianon/gosu/releases/download/1.14/gosu-armhf',
     armv7l: 'https://github.com/tianon/gosu/releases/download/1.14/gosu-armhf',
       i686: 'https://github.com/tianon/gosu/releases/download/1.14/gosu-i386',
     x86_64: 'https://github.com/tianon/gosu/releases/download/1.14/gosu-amd64'
  })
  source_sha256 ({
    aarch64: 'abb1489357358b443789571d52b5410258ddaca525ee7ac3ba0dd91d34484589',
     armv7l: 'abb1489357358b443789571d52b5410258ddaca525ee7ac3ba0dd91d34484589',
       i686: 'fa12da9c07783fe5fb2c2a81b5d4277eec7615c957a0b0cce3f14802ff9372e7',
     x86_64: 'bd8be776e97ec2b911190a82d9ab3fa6c013ae6d3121eea3d0bfd5c82a0eaf8c'
  })

  no_compile_needed

  def self.install
    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/bin"
    FileUtils.install Dir['gosu-*'], "#{CREW_DEST_PREFIX}/bin/gosu", mode: 0o755
  end
end
```
With the changes above, I get:
```
$ crew install gosu
gosu: Simple Go-based setuid+setgid+setgroups+exec
https://github.com/tianon/gosu
Version: 1.14
License: Apache-2.0
Performing pre-flight checks...
No precompiled binary available for your platform, downloading source...
[##############################################################################################################################################################################################]   2.18 MB 100%
gosu-amd64: File downloaded.
Preconfiguring package...
/usr/local/lib64/ruby/3.1.0/fileutils.rb:1395:in `initialize': No such file or directory @ rb_sysopen - /usr/local/tmp/crew/dest/usr/local/bin/gosu/gosu-amd64 (Errno::ENOENT)
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1395:in `open'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1395:in `block in copy_file'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1394:in `open'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1394:in `copy_file'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:504:in `copy_file'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:868:in `block in install'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1577:in `block in fu_each_src_dest'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1586:in `block in fu_each_src_dest0'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1584:in `each'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1584:in `fu_each_src_dest0'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:1575:in `fu_each_src_dest'
	from /usr/local/lib64/ruby/3.1.0/fileutils.rb:864:in `install'
	from /usr/local/lib/crew/packages/gosu.rb:26:in `install'
	from /usr/local/bin/crew:927:in `block in build_and_preconfigure'
	from /usr/local/bin/crew:901:in `chdir'
	from /usr/local/bin/crew:901:in `build_and_preconfigure'
	from /usr/local/bin/crew:1399:in `install'
	from /usr/local/bin/crew:1307:in `resolve_dependencies_and_install'
	from /usr/local/bin/crew:1791:in `block in install_command'
	from /usr/local/bin/crew:1786:in `each'
	from /usr/local/bin/crew:1786:in `install_command'
	from /usr/local/bin/crew:1908:in `<main>'
```

Reply:
Oops, it should be `FileUtils.install Dir['gosu-*'][0], "#{CREW_DEST_PREFIX}/bin/gosu", mode: 0o755`😅

Reply:
Nice, changes made.